client/gamescreen.js

```diff
@@ -0,0 +1,106 @@
+(function() {
+    'use strict';
+
+    var Game = window.Game = window.Game || {};
+
+    //
+    // The main state holding class for the game.
+    // It deals with input and rendering.
+    //
+
+    // Create a game screen.
+    // The startingState must be a world state message.
+    // The socket must be of type Game.Socket.
+    Game.GameScreen = function(playerId, playerSize, bulletSize, startingState, socket) {
+        var state = startingState;
+        var keymap = {};
+        var self = this;
+
+        // Get the current player entity or null if you are dead.
+        var getPlayer = function() {
+            var temp = state.alivePlayers.filter(function(player) {
+                return player.id === playerId;
+            });
+
+            return temp[0] || null;
+        }
+
+        // Update your movement vector in correspondance to the currently pressed keys.
+        var updateMoving = function(){
+            var move_x = 0;
+            var move_y = 0;
+
+            if (keymap[Game.INPUT_MOVE_UP]) {
+                move_y += -1;
+            }
+            if (keymap[Game.INPUT_MOVE_DOWN]) {
+                move_y += 1;
+            }
+            if (keymap[Game.INPUT_MOVE_LEFT]) {
+                move_x += -1;
+            }
+            if (keymap[Game.INPUT_MOVE_RIGHT]) {
+                move_x += 1;
+            }
+
+            socket.startMoving(Game.makeVector(move_x, move_y));
```

Comment:
If we don't send a stop moving message, server should presume the player is still moving in the direction they last said.